lexicon/providers/corenetworks.py

```diff
@@ -0,0 +1,280 @@
+        payload = self._get("/dnszones/{0}/records/".format(self.domain), query_params)
+        for record in payload:
+            record['content'] = record.pop('data')
+            record['name'] = self._full_name(record['name'])
+            # Core Networks' API does not provide unique IDs for each record
+            # so we generate them ourselves.
+            record['id'] = self._make_identifier( rtype = record['type'], name = record['name'], content = record['content'] )
+        return payload
+
+    def _create_record(self, rtype, name, content):
+        """Creates a record. If record already exists with the same content, do nothing."""
+
+        # Check for existence of record.
+        existing_records = self._list_records(rtype, name, content)
+        new_record_id = self._make_identifier(rtype, self._full_name(name), content)
+        record = next((r for r in existing_records if r["id"] == new_record_id), None)
+        # Nothing to do if true.
+        if record:
+            return True
+
+        data = {
+            'name': self._relative_name(name),
+            'data': content,
+            'type': rtype
+        }
+        if self._get_lexicon_option('ttl'):
+            data['ttl'] = self._get_lexicon_option('ttl')
+        if self._get_lexicon_option('priority'):
+            data['priority'] = self._get_lexicon_option('priority')
+
+        payload = self._post("/dnszones/{0}/records/".format(self.domain), data)
+        # Changes to the zone need to be committed.
+        # https://beta.api.core-networks.de/doc/#functon_dnszones_commit
+        self.modified = True
+
+        return new_record_id
+
+    def _update_record(self, identifier, rtype=None, name=None, content=None):
+        """Updates a record. Core Networks neither supports updating a record nor is able to reliably identify a record
+        after a change. The best we can do is to identify the record by ourselves, fetch its data, delete it and
+        re-create it."""
+        if identifier is not None:
+            # Check for existence of record
+            existing_records = self._list_records(rtype)
+            record = next((r for r in existing_records if r["id"] == identifier), None)
+            if not record:
+                return True
+            if rtype:
+                record['type'] = rtype
+            if name:
+                record['name'] = self._relative_name(name)
+            if content:
+                record['content'] = content
+            if self._delete_record(identifier):
+                new_id = self._create_record(rtype = record['type'], name = record['name'], content = record['content'])
+                return new_id
+        else:
+            records = self._list_records( rtype=rtype, name=self._relative_name(name) )
+            if len(records) > 0:
+                for record in records:
+                    return self._update_record( record['id'], rtype, name, content )
```

Comment:
Mh, haven't tried out, but looks like only the first match will be updated, right?

Reply:
This is a design decision here: either you update all matching records, or only the first one.

Usually we update the first one, but generate a log warning to say that more than one record was found, and only first one will be updated.